src/Symfony/Component/Mailer/Bridge/Mailjet/Transport/MailjetApiTransport.php

```diff
@@ -85,7 +85,7 @@ protected function doSendApi(SentMessage $sentMessage, Email $email, Envelope $e
             throw new HttpTransportException(sprintf('Unable to send an email: "%s" malformed api response.', $response->getContent(false)), $response);
         }
 
-        $sentMessage->setMessageId($response->getHeaders(false)['x-mj-request-guid'][0]);
+        $sentMessage->setMessageId((string) $result['Messages'][0]['To'][0]['MessageID'] ?? '');
```

Comment:
I don't think the cast to string is correct as it applies before the `??`
Also why is it needed at all?

Reply:
That was a remnant from earlier, since HttpClient uses JSON_BIGINT_AS_STRING, it's no longer needed.